src/main/java/com/amida/saraswati/edifhir/controller/X12EDIController.java

```diff
@@ -111,6 +139,45 @@ public static X12DATA_TYPE getType(String name) {
         }
     }
 
+    @PostMapping("/poststream")
+    public ResponseEntity<String> postmessage(
+            @RequestBody String data, @RequestParam(name = "topic") String topic
+    ) {
+        log.info("postmessage to {}", Optional.ofNullable(topic).orElse("Not Provided!"));
+        try {
+            if (consumerTopic.equals(topic) || publishTopic.equals(topic)) {
+                streamService.publishMessage(topic, messageKey, data);
+                return ResponseEntity.ok("message posted");
+            } else {
+                return ResponseEntity.badRequest().body("Unsupported topic.");
+            }
+        } catch (StreamException e) {
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(e.getMessage());
+        }
+    }
+
+    @GetMapping("/getstreammessage")
+    public ResponseEntity<String> getMessage(
+            @RequestParam(name = "topic") String topic,
+            @RequestParam(required = false, name = "isCommitting") boolean isCommitting
+    ) {
+        log.info("getstreammessage from {}", Optional.ofNullable(topic).orElse("Not Provided!"));
+        try {
+            List<EdiFhirMessage> result = streamService.pollMessage(topic, isCommitting);
+            StringBuilder msg = new StringBuilder();
+            String headline = String.format("Total messages received from the topic, %s: %d\n",
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.